file_writer.go

```diff
@@ -211,7 +211,7 @@ func (f *FileWriter) Close() error {
 	}
 
 	// Retry on Complete request returning false
-	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
```

Comment:
it seems that you're using this context just as a timer, which is a bit of an overkill and puzzling.

Reply:
Agreed it's a bit overkill. I've replaced the context with a timer.